src/main/java/frc/robot/subsystems/conveyor/commands/ConveyCargo.java

```diff
@@ -0,0 +1,61 @@
+package frc.robot.subsystems.conveyor.commands;
+
+import edu.wpi.first.wpilibj.Timer;
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import frc.robot.Constants;
+import frc.robot.RobotContainer;
+import frc.robot.subsystems.conveyor.Conveyor;
+
+public class ConveyCargo extends CommandBase {
+    private final Conveyor conveyor;
+    private final Timer timer = new Timer();
+    private boolean stopForSecondBall = false;
+    private boolean lastPreFlapObservation = false;
+
+    public ConveyCargo(Conveyor conveyor) {
+        this.conveyor = conveyor;
+        addRequirements(conveyor);
+    }
+
+    @Override
+    public void initialize() {
+        timer.reset();
+        timer.stop();
+    }
+
+    @Override
+    public void execute() {
```

Comment:
I'm not sure it would work the way you wanted (and I certain that it can be implemented in an easier way)

Reply:
What wouldn't work and what's not easy about it? The first ball I shoot straight away. As soon as I see a ball went in and out of the pre flap or into the post flap I know the second ball is next, and then I wait only for it at the pre flap for 0.3 seconds